src/block_cache/block_chunkcache.c

```diff
@@ -184,75 +184,64 @@ __chunkcache_make_hash(WT_CHUNKCACHE *chunkcache, WT_CHUNKCACHE_HASHID *hash_id,
-            found_eviction_candidate = true;
-            break;
-        }
-    }
-    __wt_spin_unlock(session, &chunkcache->chunkcache_lru_lock);
-
-    if (!found_eviction_candidate)
-        return;
-
-    __wt_verbose(session, WT_VERB_CHUNKCACHE, "evict: %s(%u), offset=%" PRIu64 ", size=%" PRIu64,
-      (char *)&chunk_to_evict->hash_id.objectname, chunk_to_evict->hash_id.objectid,
-      (uint64_t)chunk_to_evict->chunk_offset, (uint64_t)chunk_to_evict->chunk_size);
-
-    __wt_spin_lock(session, WT_BUCKET_LOCK(chunkcache, chunk_to_evict->bucket_id));
-    TAILQ_REMOVE(
-      WT_BUCKET_CHUNKS(chunkcache, chunk_to_evict->bucket_id), chunk_to_evict, next_chunk);
-    __wt_spin_unlock(session, WT_BUCKET_LOCK(chunkcache, chunk_to_evict->bucket_id));
-    __chunkcache_free_chunk(session, chunk_to_evict);
-
-    WT_STAT_CONN_INCR(session, chunk_cache_chunks_evicted);
+    /* Do not evict chunks that are in the processed of being added to the cache. */
+    if (!chunk->valid)
+        return (false);
+    if (--(chunk->access_count) == 0)
+        return (true);
+    return (false);
 }
 
 /*
  * __chunkcache_eviction_thread --
- *     Periodically sweep the cache and evict chunks at the end of the LRU list.
+ *     Periodically sweep the cache and evict chunks with a low access count. This is similar to the
+ *     clock eviction algorithm, which is an approximation of the LRU algorithm.
  */
 static WT_THREAD_RET
 __chunkcache_eviction_thread(void *arg)
 {
     WT_CHUNKCACHE *chunkcache;
+    WT_CHUNKCACHE_CHUNK *chunk, *chunk_tmp;
     WT_SESSION_IMPL *session;
+    int i;
 
     session = (WT_SESSION_IMPL *)arg;
     chunkcache = &S2C(session)->chunkcache;
 
     while (!chunkcache->chunkcache_exiting) {
-        /* Try evicting a chunk if we have exceeded capacity. */
-        while (!chunkcache->chunkcache_exiting &&
-          ((chunkcache->bytes_used + chunkcache->chunk_size) >
-            chunkcache->evict_trigger * chunkcache->capacity / 100))
-            __chunkcache_evict_one(session);
-        __wt_sleep(0, 100 * WT_THOUSAND); /* may need tuning */
+        /* Do not evict if we are not close to exceeding capacity */
+        if ((chunkcache->bytes_used + chunkcache->chunk_size) <
+          chunkcache->evict_trigger * chunkcache->capacity / 100) {
+            __wt_sleep(1, 0);
+            continue;
+        }
+        for (i = 0; i < (int)chunkcache->hashtable_size; i++) {
+            __wt_spin_lock(session, &chunkcache->hashtable[i].bucket_lock);
+            TAILQ_FOREACH_SAFE(chunk, WT_BUCKET_CHUNKS(chunkcache, i), next_chunk, chunk_tmp)
+            {
+                if (__chunkcache_should_evict(chunk)) {
+                    TAILQ_REMOVE(WT_BUCKET_CHUNKS(chunkcache, i), chunk, next_chunk);
+                    __chunkcache_free_chunk(session, chunk);
+                    WT_STAT_CONN_INCR(session, chunk_cache_chunks_evicted);
+                    __wt_verbose(session, WT_VERB_CHUNKCACHE,
```

Comment:
I think this should be `debug2`, it's potentially a lot of noise. (I realise now that the other messages with similar frequency should also be `debug2` - that can be addressed during the project, I won't ask you to fix those right now.)